### PR TITLE
ci: update m4mkdocs with insiders version.

### LINKED ISSUE #247: [M4M | BLOG] Updated kernl/docs CI to support M4M Insiders.
- [x] Add the secret repository `GH_TOKEN` (value of the personal token created by @pommedeterresautee and stored in bitwarden) to allow the pull of the Insiders repository.
- [x] Change the 'deploy-static-site.yml' workflow to pull the Insiders repository.
- [x] Changing the docker file to pull the image from the private registry.

> Please note

- That we don't use the [inheritance configuration](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#built-in-plugins) to avoid complexity at this stage.
- We target a specific version to avoid breaking changes.

### PR DESCRIPTION
**Pull request by white-gorilla**
This PR modifies the CI to use the Insiders version of Material for MkDocs.

- [x] Add the secret repository `GH_TOKEN` (value of the personal token created by @pommedeterresautee and stored in bitwarden) to allow the pull of the Insiders repository.
- [x] Change the 'deploy-static-site.yml' workflow to pull the Insiders repository.
- [x] Changing the docker file to pull the image from the private registry.

> Please note

- That we don't use the [inheritance configuration](https://squidfunk.github.io/mkdocs-material/insiders/getting-started/#built-in-plugins) to avoid complexity at this stage.
- We target a specific version to avoid breaking changes.

fix #247

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
+run: pip install git+https://${{secrets.GH_TOKEN}}@github.com/squidfunk/mkdocs-material-insiders.git@9.0.5-insiders-4.28.0 "mkdocstrings[python]"
+```
+
+### Version upgrade
+
+- Check the [changelog](https://squidfunk.github.io/mkdocs-material/insiders/changelog/) to verify the current version and take into account the breaking changes if any. 
+- Synchronize the fork manually (@pommedeterresautee has the rights).
+- Get the version number from [the changelog](https://github.com/ELS-RD/mkdocs-material-insiders/blob/master/CHANGELOG) of the repo.
+  - Ex.: `mkdocs-material-9.0.5+insiders-4.28.0` (it must respect this pattern).
+- Create a release manually with :
```
Comment (jonathlela) [inline docs/README.md:29]: I don't understand this part. what is creating a release manually ?

File: docs/README.md [hunk 2]
```diff
@@ -1,5 +1,46 @@
 # Kernl static documentation site
 
+## Targeting the Material for MkDocs Insiders version
```
Comment (jonathlela) [inline docs/README.md:3]: I'm not fond of telling in a public repository the name of our token, how we get our private version of m4mk and a way to create automaticly a new image in our private registry. Shall we make this part minimal (just the part with the version number and the warnings before upgrading), and put the full way of using m4mk in our blog repo ?

**Review by jonathlela — COMMENTED** [review]
I have some security issue for this PR.

File: docs/README.md (hunk 1 above)
Comment (white-gorilla) [reply docs/README.md:29]: Whether the fork is manually or automatically synchronized, you have to do the release yourself. The M4M maintainer suggests that we contribute by working on a specific workflow.

While waiting to plan, or not, this work must be done via the interface. Maybe this is what should be specified?

Capture d’écran 2023-01-19 à 09 48 44

File: docs/README.md (hunk 2 above)
Comment (white-gorilla) [reply docs/README.md:3]: @jonathlela You are absolutely right. I will correct.

What do you mean by blog repo?

Comment (white-gorilla) [reply docs/README.md]: Done. Ready for review.

File: docs/README.md [hunk 3]
```diff
@@ -1,16 +1,36 @@
 # Kernl static documentation site
 
-## Preview the static site locally 
+## Material for MkDocs Insiders version
+
+To avoid any breaking change, we target a specific version.
+
+For the preview (`doc/Dockerfile.mkdocstrings`),
```
Comment (jonathlela) [inline docs/README.md:7]: `docs` not `doc`, can we make direct links ?

**Review by jonathlela — APPROVED** [review]
Just a typo and good for me.

**Pull request merged**